SingularityService/src/main/java/com/hubspot/singularity/data/history/HistoryJDBI.java

```diff
@@ -153,4 +158,58 @@ private void addWhereOrAnd(StringBuilder sqlBuilder, boolean shouldUseWhere) {
     return query.list();
   }
 
+  public int getTaskIdHistoryCount(Optional<String> requestId, Optional<String> deployId, Optional<String> host,
+      Optional<ExtendedTaskState> lastTaskStatus, Optional<Long> startedBefore, Optional<Long> startedAfter) {
+
+    final Map<String, Object> binds = new HashMap<>();
+    final StringBuilder sqlBuilder = new StringBuilder(GET_TASK_ID_HISTORY_COUNT_QUERY);
+
+    if (requestId.isPresent()) {
```

Comment:
rather than rewriting all of this from the method above, maybe make it one private method that get... and get..Count can call, just with different SELECT statements at the front?